### PR TITLE
Feat(#68): Shop bloc

### LINKED ISSUE #68: 👨‍💻 Codigin: creare `Bloc` dello shop con `Milestone` e lista di `Reward`
# Descrizione
Creare Bloc con Stato ed Eventi associati allo shop

### Epic:

- [ ] #3 

## Dettagli tecnici e analisi:
Eventi:
EventSetMilestone
EventNewReward
EventDeleteReward

Stato:
List<Reward, index milestone (o primo reward)

> Aggiungi reference utili a chi dovrà eseguire il task (es. nomi file, nomi classi ecc..)

## Acceptance Test:

> Aggiungi qui tutti test necessari per verificare il corretto funzionamento della nuova feature

- [ ] Happy flow
- [ ] Gestione errori
- [ ] Edge cases
- [ ] ...

### PR DESCRIPTION
**Pull request by gallottino**
## Related to

Closes #68 

- [ ] #68

### Dettagli aggiuntivi
Alla fine ho messo il milestone come Reward direttamente perché come indice non sarebbe stata coerente nel caso in cui qualcuno eliminava una reward dallo shop

### CODE REVIEW

File: lib/models/shop/shop.dart [hunk 1]
```diff
@@ -1,17 +1,43 @@
+import 'package:equatable/equatable.dart';
 import 'package:json_annotation/json_annotation.dart';
 import 'package:kaizen/models/common/reward/reward.dart';
 part 'shop.g.dart';
 
 @JsonSerializable()
-class Shop {
-  final Reward milestone;
+class Shop extends Equatable {
+  final Reward? milestone;
```
Comment (gallottino) [inline lib/models/shop/shop.dart:8]: La milestone in questo caso è un `Nullable Reward` perché altrimenti se un utente elimina dallo shop una Reward l'indice si sfalza e in questo modo non gestisco nemmeno sorting. MI sembrava soluzione migliore, se avete altre idee suggerite pure, ma non credo cambi troppo.

**Review by Oniryu95 — APPROVED** [review]

File: lib/models/shop/shop.dart (hunk 1 above)
Comment (Oniryu95) [reply lib/models/shop/shop.dart:8]: Effettivamente non ci abbiamo pensato ieri. La milestone come Reward? per me è buona soluzione.

**Review by Hugene — APPROVED** [review]
